### PR TITLE
All changes

### PR DESCRIPTION
**Pull request by Antrakos**
- Refactoring
- MVEL templates
- Handlebars templates

### CODE REVIEW

**Comment by Antrakos** [comment]
You should delete all files in repo that are ignored by Git

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +0,0 @@
-#Mon Mar 07 21:06:45 EET 2016
-distributionBase=GRADLE_USER_HOME
-distributionPath=wrapper/dists
-zipStoreBase=GRADLE_USER_HOME
-zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.12-all.zip
```
Comment (Mart-Bogdan) [inline gradle/wrapper/gradle-wrapper.properties:6]: Why did you removed gradle-wrapper, how am I supposed to build project now?

File: jmh-results.txt [hunk 2]
```diff
@@ -1,9 +0,0 @@
-Benchmark                                   Mode  Cnt       Score       Error  Units
-Jade.simple_template                       thrpt   25  356691.343 ±  2944.519  ops/s
-Jade.with_layout                           thrpt   25  361213.392 ±  4055.455  ops/s
-JadeNoCompressedOops.simple_template       thrpt   25  318515.394 ± 34091.816  ops/s
-JadeNoCompressedOops.with_layout           thrpt   25  333746.770 ±  4027.133  ops/s
-Thymeleaf.simple_template                  thrpt   25   66480.184 ±  1188.853  ops/s
-Thymeleaf.with_layout                      thrpt   25   86669.079 ±  1748.425  ops/s
-ThymeleafNoCompressedOops.simple_template  thrpt   25   59239.638 ±   593.229  ops/s
-ThymeleafNoCompressedOops.with_layout      thrpt   25   82978.264 ±  1315.709  ops/s
```
Comment (Mart-Bogdan) [inline jmh-results.txt:9]: This file was especially placed in this dir, manually copied from build/ to be present in repo, with latest results.

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (Antrakos) [inline gradle/wrapper/gradle-wrapper.properties:6]: "gradle build" without any unnecessary wrappers

File: jmh-results.txt (hunk 2 above)
Comment (Antrakos) [inline jmh-results.txt:9]: My bad. I'll get it back.

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (Mart-Bogdan) [inline gradle/wrapper/gradle-wrapper.properties:6]: You don't get the pint, if I use wrapper:
1. I don't need to install gradle at my system at all
2. Build didn't depend on version of gradle installed locally, each build exactly specify which version of gradle is to be used

That's best practices

**Pull request merged**